performance/shared/src/main/kotlin/org/jetbrains/benchmarksLauncher/launcher.kt

```diff
@@ -25,13 +25,18 @@ abstract class Launcher(val numWarmIterations: Int, val numberOfAttempts: Int, v
 
     abstract val benchmarks: BenchmarksCollection
 
-    protected val benchmarkResults = mutableListOf<BenchmarkResult>()
-
-    fun launch(benchmarksToRun: Collection<String>? = null): List<BenchmarkResult> {
-        val runningBenchmarks = benchmarksToRun ?: benchmarks.keys
-        runningBenchmarks.forEach {
-            val benchmark = benchmarks[it]
-            benchmark ?: error("Benchmark $it wasn't found!")
+    fun launch(filters: Collection<String>? = null,
+               filterRegexes: Collection<String>? = null): List<BenchmarkResult> {
+        val regexes = filterRegexes?.map { it.toRegex() } ?: listOf()
+        val filterSet = filters?.toHashSet() ?: hashSetOf()
+        // Filter benchmarks using given filters, or run all benchmarks if none were given.
+        val runningBenchmarks = if (filterSet.isNotEmpty() || regexes.isNotEmpty()) {
+            benchmarks.filterKeys { bm -> bm in filterSet || regexes.any { r -> r.matches(bm) } }
```

Comment:
As far as I know we try escape such short forms like `bm`, `r`.

Reply:
Spelled out "benchmark" and removed "r". In this case I need to give at least one of them a name. I can't use the default "it" name for both since both are used in the innermost lambda.